### PR TITLE
[Serverless] Updates serverless-init to use fxutil.OneShot

### PR DESCRIPTION
**Pull request by duncanpharvey**
### What does this PR do?

Updates serverless-init to call its entry point with fxutil.OneShot so components can be injected as dependencies.

Summary of code changes:

* Created `cliParams` struct
* Added wrapper function `run` to call `initcontainer.Run` with values returned from `setup` and `cliParams`
* `run` in serverless-init/main.go is called with fxutil.OneShot

### Motivation

https://datadoghq.atlassian.net/browse/SVLS-4183

### Describe how to test/QA your changes

* Run the Lambda Extension locally using files provided here: https://github.com/DataDog/datadog-lambda-extension/tree/main/local_tests/serverless-init
  * `./build-docker-runtime.sh`
  * `./invoke.sh 2`
  * Verify that extension loads and runs without error

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: cmd/serverless-init/main.go [hunk 1]
```diff
@@ -35,15 +38,32 @@ const (
 	loggerName        = "SERVERLESS_INIT"
 )
 
+type cliParams struct {
```
Comment (alexgallotta) [inline cmd/serverless-init/main.go:41]: Do we really need to define this type? Or is it a golang best practice?

**Review by alexgallotta — COMMENTED** [review]
Little comment but overall looks good!

File: cmd/serverless-init/main.go (hunk 1 above)
Comment (duncanpharvey) [reply cmd/serverless-init/main.go:41]: It's a convention we're following for One-Shot Fx apps. Here's the internal documentation for it: https://github.com/DataDog/datadog-agent/blob/921e96c33fb884720a6487b1c41d6ba373746adc/docs/components/defining-apps.md#one-shot-apps

**Review by alexgallotta — APPROVED** [review]

**Pull request merged**